orttraining/orttraining/python/training/ortmodule/_options.py

```diff
@@ -0,0 +1,336 @@
+        """
+
+        return not _SkipCheck.is_disabled(self) and check in self
+
+    def is_disabled(self):
+        """Check whether `_SkipCheck.SKIP_CHECK_DISABLED is set on the `_SkipCheck instance"""
+
+        return _SkipCheck.SKIP_CHECK_DISABLED in self
+
+
+class _RuntimeOptions:
+    """Configurable runtime options for ORTModule."""
+
+    def __init__(self, logger: Logger, config=None):
+        """Constructor for _RuntimeOptions.
+
+        Initially set all the options to their default values, then override them with the values
+        from the environment variables.
+        """
+        self._logger = logger
+
+        self.onnx_opset_version = ortmodule.ONNX_OPSET_VERSION
+        self.conv_algo_search = "HEURISTIC"
+
+        # Configuration for cast optimization.
+        # Specify cast propagation strategy. Currently, three strategies are available:
+        #  NONE, INSERT-AND-REDUCE and FLOOD-FILL
+        # The default is FLOOD_FILL, expand FP16 computation regions in the graph using
+        # allowed opcodes for the given level.
+        self._propagate_cast_ops_strategy = C.PropagateCastOpsStrategy.FLOOD_FILL
+        # Optimize by moving Cast operations if propagate_cast_ops_level is non-negative.
+        # - If the _propagate_cast_ops_level is set to zero, then the transformation considers only the opcodes
+        #   specified by _propagate_cast_ops_allow as "FP16 safe", to insert/(re)move cast operations before/after
+        #   to perform such operations in reduced (16-bit) precision.
+        # - If propagate_cast_ops_level is positive, 1 or 2, then in addition to opcode codes specified by
+        #   propagate_cast_ops_allow, use onnxruntime predetermined list of opcodes considered safe to move
+        #   before/after the cast operation.
+        # - Onnxruntime Level 1 predetermined "FP16 safe" opcodes include only opcodes that do not perform
+        #   any computation such as Transpose, Split, Reshape, etc., or the computation is actually in Float
+        #   such as GeLU, etc.
+        # - Whereas Level 2 predetermined "FP16 safe" opcodes include opcodes that perform computation using
+        #   contrib ops, Dropout, LayerNormalization, etc.
+        self._propagate_cast_ops_level = 1
+        # List of opcodes to be considered safe to move before/after the cast operation if propagate_cast_ops_level
+        # is zero.
+        self._propagate_cast_ops_allow = []
+
+        # default execution order is priority-based for both dynamic/static shape input for now
+        # if we observe the benefit of static shape, we can expose this flag to the user
+        self._use_static_shape = False
+
+        # flag to enable symbolic shape inference for dynamic shape inputs to improve performance
+        self._run_symbolic_shape_infer = True
+
+        # PyTorch custom Autograd function support
+        from ._custom_autograd_function import custom_autograd_function_enabler
```

Comment:
This import will always be executed with ORTModule (since it is part of the constructor). Shall we move it to the top of this file?